docs/spec.md

```diff
@@ -326,8 +326,10 @@ Informs the client about updates to the connection graph. This is only sent to c
 - `channels`: array of:
   - `id`: number chosen by the client. The client may reuse ids that have previously been unadvertised.
   - `topic`: string
-  - `encoding`: string, one of the encodings [supported by the server](#server-info)
+  - `encoding`: string, one of the message encodings [supported by the server](#server-info)
   - `schemaName`: string
+  - `schema`: string | undefined, optional [schema definition](#advertise).
+  - `schemaEncoding`: string | undefined, optional [schema encoding](#advertise).
```

Comment:
Should we say schemaEncoding is required if schema is set?

Reply:
:+1: 
Is there a way to enforce this with typescript?

Reply:
You might have to do something like `{ ...; schema: string; schemaEncoding: string; } | { ...; schema?: undefined; schemaEncoding?: undefined }`

Reply:
I tried this, but this doesn't work well together with `Omit`: https://www.typescriptlang.org/play?#code/C4TwDgpgBAwgNgSwgO2DAFgQ2cicoC8UA3gFBRQIAmAXFMgK4C2ARhAE4Dc5UwA9mAQBjOgGdg7BMgDm3CiiF8qU6WIkq5UUUPQQmmAHKYmENZJncAvlABkUABTEtOvZjMbnu-QFFki5TLuMlDWAD4knq50DMhUEABmUhBUnJE+fkoq0bEJSVQhAJTcpKTxMULACHzIUJhUAG4claIQ9kKIKGhYOHhQdADyTAjAADzwSKgY2LhwADRQAETUCwB8BSQ8FFCKyOKU+UQAjJpbO3u4AO5TPfh0453XM4QRAHRv7RNd03jz1CGpmwoZz4cAgLzgfGk9kujzwRVIliAA